test/e2e/e2e_test.go

```diff
@@ -135,3 +137,43 @@ func TestMain(m *testing.M) {
 func TestE2E(t *testing.T) {
 	RunE2ETests(t)
 }
+
+var _ = ginkgo.ReportAfterSuite("Kubernetes e2e suite report", func(report ginkgo.Report) {
+	var f *os.File
+	var err error
+	// The DetailsRepoerter will output details about every test (name, files, lines, etc) which helps
+	// when documenting our tests.
+	if len(framework.TestContext.SpecSummaryOutput) > 1 {
+		if _, err = os.Stat(framework.TestContext.SpecSummaryOutput); os.IsNotExist(err) {
+			f, err = os.Create(framework.TestContext.SpecSummaryOutput)
+			if err != nil {
+				klog.Errorf("Failed to create reporter: %v", err)
+				return
+			}
+		} else {
+			f, _ = os.OpenFile(framework.TestContext.SpecSummaryOutput, os.O_RDWR|os.O_APPEND, 0660)
+		}
+
+		for _, specReport := range report.SpecReports {
+			if f != nil {
```

Comment:
if we handle the errors we should not have to check if is nil

Reply:
Done